### ISSUE TITLE
Inconsistant values of lr_scheduler.get_lr and lr in optimizer.param_groups

### ISSUE THREAD

**Issue by lihuanglx** [pytorch/pytorch#20138]
## 🐛 Bug

After upgrading to 1.1.0, the value returned by `lr_scheduler.get_lr` is confusing comparing to the value inside `optimizer.param_groups`.

## To Reproduce

Code:

```python
import torch

net = torch.nn.Conv2d(1, 1, 1)
optimizer = torch.optim.SGD(net.parameters(), lr=0.1)
lr_scheduler = torch.optim.lr_scheduler.ExponentialLR(optimizer, gamma=0.9)

for i in range(5):
    print(i, lr_scheduler.get_lr(), optimizer.param_groups[0]['lr'])
    lr_scheduler.step()
```

Output:

```
0 [0.1] 0.1
1 [0.08100000000000002] 0.09000000000000001
2 [0.07290000000000002] 0.08100000000000002
3 [0.06561000000000002] 0.07290000000000002
4 [0.05904900000000002] 0.06561000000000002
```

We got inconsistant values. The two values are the same in the first line, but different in following lines by lr decay factor `gamma = 0.9`. 

## Expected behavior

At least we should have consistant values of the two, right?

## Environment

I'll skip this part since it can be easily reproduced from a fresh 1.1.0 installation.

**Comment by codexetreme** [comment]
If this is a bug, can I try to work on it?

**Comment by vfdev-5** [comment]
@lihuanglx I would say this is due to recursive reimplementation of LR scheduler, https://github.com/pytorch/pytorch/pull/14010
